File: src/js/util/printing.js
/*
  loricarlos@gmail.com
*/

/*
  Utility for printing with some margin
*/
var margin = 0,
    increment = 3,
    marginChar = ' ',
	verbose = false
;
function resetMargin(){
	margin = 0;
	increment = 3,
    marginChar = ' '
}
function incMargin(){
	margin += increment;
}

function decMargin(){
	margin -= increment;
}
function printMargin(s){
	if (! verbose ) return;
	for (var i = 1; i <= margin; i++) 
		write(marginChar);
	print(s);
}

function printPlain(s){
	if (! verbose ) return;
	print(s);
}

File: src/java/util/Printing.java
import java.util.*;
package eif203.labs.permutate;


public static class Printing{
	
	int margin = 0,
    int increment = 3,
    char marginChar = ' ',
	boolean verbose = false
;
public static void resetMargin(){
	margin = 0;
	increment = 3,
    marginChar = ' '
}
public static void incMargin(){
	margin += increment;
}

public static void decMargin(){
	margin -= increment;
}
public static void printMargin(String s){
	if (! verbose ) return;
	for (int i = 1; i <= margin; i++) 
	System.out.print(marginChar);
	System.out.println(s);
}

public static void printPlain(String s){
	if (! verbose ) return;
	System.out.println(s);
}

}